operatorcourier/validate.py

```diff
@@ -184,79 +184,180 @@ def _csv_spec_validation(self, spec, bundleData):
             valid = False
 
         if "customresourcedefinitions" in spec:
-            customresourcedefinitions = spec["customresourcedefinitions"]
+            if self._csv_crd_validation(
+                    spec["customresourcedefinitions"], bundleData) is False:
+                valid = False
+
+        if "apiservicedefinitions" in spec:
+            if self._csv_asd_validation(spec["apiservicedefinitions"]) is False:
+                valid = False
+
+        return valid
+
+    def _csv_crd_validation(self, customresourcedefinitions, bundleData):
+        valid = True
+
+        crdList = []
+        for crd in bundleData[self.crdKey]:
+            try:
+                name = crd["metadata"]["name"]
+                crdList.append(name)
+            except KeyError:
+                pass
+
+        if "owned" not in customresourcedefinitions:
+            self._log_error("spec.customresourcedefinitions.owned"
```

Comment:
Not all operators that have the `customresourcedefinitions` field operate on `"owned"` CRDs. I believe this should be an early exit rather than a logged error

Reply:
I agree that having `customresourcedefinitions` without any subfields should raise an error. But there are other values than `"owned"` that could exist here (such as `required`: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#required-crds)

Reply:
> I agree that having `customresourcedefinitions` without any subfields should raise an error.

To contradict myself, what's the benefit of enforcing that subfields exist?